src/components/Recorder/Recorder.tsx

```diff
@@ -48,14 +47,23 @@ export class Recorder extends Component<Props> {
     }
 
     componentDidUpdate(prevProps: Props) {
-        const { doRecording, discard } = this.props;
+        const { doRecording, discard, screenStream, videoStream } = this.props;
 
         if (discard) {
             this.recordedBlobs = [];
             this.screenBlobs = [];
+            this.mediaRecorder = undefined;
+            this.screenRecorder = undefined;
             return;
         }
 
+        if (!videoStream) {
+            this.screenRecorder = undefined;
```

Comment:
@Channa53  are you sure you want here to reset the 
            this.screenRecorder = undefined;
 ?

Reply:
yes, it's neing used only while recording. and during that time no props or states should be updated

Reply:
ah you right I have an issue here..

Reply:
fixed